content/resources/developer/microservice.md

```diff
@@ -0,0 +1,69 @@
+
+### FENCE
+The Fence service controls access to the metadata, submission, indexing, and data itself. Fence is an authentication (AuthN) and authorization (AuthZ) service which utilizes OpenID Connect flow (an extension of OAuth2) to generate tokens for clients. It can also provide tokens directly to a user. Clients and users may then use those tokens (JWT) with other Gen3 Data Commons services to access protected endpoints that require specific permissions. Fence can be configured to support different Identity Providers (IDPs) for AuthN. At the moment, supported IDPs include Google, and Shibboleth supporting providers such as NIH iTrust.
+
+
+### INDEXD
+The Indexd service provides permanent digital IDs for data objects. These IDs can be used to retrieve the data, or query the metadata associated with the object. The Indexd service tracks the locations and hash of every asset (file) in the data commons object store. It exports RESTful APIs for registering a new asset, and retrieving data for an existing asset.
+
+### WINDMILL
+The Windmill service is an interactive website that allows users to explore, submit, and download data. The Windmill service utilizes the APIs offered by the data commons just as any other externally built app could.
+
+### Microservice NGINX Route Table
+
+This table is helpful for debugging errors in front-end applications like [Windmill: data portal](https://github.com/uc-cdis/data-portal) or other Gen3 clients. You can easily identify the running service that is returning an error, based on its absolute HTTP request path. [Source](https://github.com/uc-cdis/cloud-automation/tree/master/kube/services/revproxy/gen3.nginx.conf).
+
+| Microservice                 | URL Path Prefix (NGINX Location) | GitHub Repository                                                                          |
+|------------------------------|----------------------------------|--------------------------------------------------------------------------------------------|
+| ambassador-service           | /lw-workspace/proxy/             | https://github.com/uc-cdis/cloud-automation/tree/master/kube/services/ambassador           |
+| arborist-service             | /gen3-authz                      | https://github.com/uc-cdis/arborist                                                        |
+| arborist-service             | ~ /authz/?                       | https://github.com/uc-cdis/arborist                                                        |
+| arborist-service             | /authz/resources                 | https://github.com/uc-cdis/arborist                                                        |
+| arborist-service             | /gen3-authz-test                 | https://github.com/uc-cdis/arborist                                                        |
+| arranger-service             | /api/v0/flat-search/             | https://github.com/uc-cdis/arborist                                                        |
```

Comment:
This says "arranger-service" but the GitHub repo is "Arborist"
we're also encouraging users not to use Arranger so maybe we can get rid of it?

Reply:
Removed